File: backend/src/main/java/model/Company.java
package model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("companies")
public record Company(
    @Id
    String id,
    String name,
    List<User> employees,
    String email,
    String phoneNumber,
    List<User> customers,
    List<Product> product,
    Address address
) {
}

File: frontend/src/components/model/Company.ts
import {Address} from "./Address";
import {Product} from "./Product";
import {User} from "./User";

export type Company = {
    name: string,
    employees: User[],
    email: string,
    phoneNumber: string,
    customers: User[],
    products: Product[],
    address: Address
}
